ql/experimental/math/hybridsimulatedannealing.hpp

```diff
@@ -87,7 +87,7 @@ namespace QuantLib {
                                  ResetScheme resetScheme = ResetToBestPoint,
                                  Size resetSteps = 150,
                                  ext::shared_ptr<OptimizationMethod> localOptimizer =
-                                     ext::shared_ptr<OptimizationMethod>(),
+                                     ext::shared_ptr<OptimizationMethod>(new LevenbergMarquardt),
```

Comment:
Just writing this as a suggestion so I can apply it directly from the PR without bothering you any longer:
```suggestion
                                     ext::make_shared<LevenbergMarquardt>(),
```

Reply:
I wonder if we can get clang-tidy to do this for us. There's a lot of code that does `shared_ptr(new Foo)` instead of `make_shared<Foo>()`.

Reply:
Sometimes it does (it converted a bunch of those when we first started using it), sometimes it doesn't.  I'm not sure I fathom why.  I'm under the impression that it detects `shared_ptr<T>(new T)` but it struggles with `shared_ptr<Base>(new Derived)`, but I might be wrong.

Reply:
Oh I see. The latter is not always safe to convert. Maybe they didn't implement a more detailed analysis.